backend/src/main.py

```diff
@@ -28,6 +32,55 @@ async def lifespan(app: FastAPI):
     allow_headers=["*"],
 )
 
+security = HTTPBearer()
+
+
+@app.middleware("http")
+async def validate_token(request: Request, call_next):
+    if request.url.path == "/docs" or request.url.path == "/openapi.json":
+        return await call_next(request)
+
+    if os.getenv("TESTING", False):  # Skip validation in tests
+        return await call_next(request)
+
+    if "Authorization" not in request.headers:
+        raise HTTPException(status_code=403, detail="Authorization header missing")
+
+    auth: HTTPAuthorizationCredentials = await security(request)
+    token = auth.credentials
+
+    if token != settings.bearer_token:
+        raise HTTPException(status_code=403, detail="Invalid token")
+
+    return await call_next(request)
+
+
+def custom_openapi():
+    if app.openapi_schema:
+        return app.openapi_schema
+    openapi_schema = get_openapi(
+        title="NoName Anonymous Surveys API",
+        version="1.0.0",
+        description="This is the API for the NoName Anonymous Surveys project.",
+        routes=app.routes,
+    )
+    security_scheme = {
+        "HTTPBearer": {
+            "type": "http",
+            "scheme": "bearer",
+            "bearerFormat": "JWT",
+        }
+    }
+    openapi_schema["components"]["securitySchemes"] = security_scheme
+    for path in openapi_schema["paths"].values():
+        for method in path.values():
+            method["security"] = [{"HTTPBearer": []}]
```

Comment:
I'll try to make it better later, please don't close the task on jira and instead assign it to me